### PR TITLE
Issue #9756: IllegalIdentifierName doesn't check single lambda parameter

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #9756: IllegalIdentifierName doesn't check single no-parentheses lambda parameter

Diff Regression config: https://gist.githubusercontent.com/nmancus1/2267f758c41c127dfee6c6c73c655b0c/raw/c1a31b23cd28a0a476b094790fbf034c26757e01/base.xml

Diff Regression projects: https://gist.githubusercontent.com/nmancus1/a89d2614131e8c0a44f3b31cb6775598/raw/c9476797246225ab23ce707662e1e6b9ce2caa4e/gistfile1.txt

### CODE REVIEW

**Comment by nmancus1** [comment]
Github, generate report

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/illegalidentifiername/InputIllegalIdentifierNameLambda.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        THU,
+        FRI,
+        SAT,
+        SUN,
+    }
+
+    int yield(Day day) { // violation
+        return switch (day) {
+            case MON, TUE -> Math.addExact(0, 1);
+            case WED -> Math.addExact(1, 1);
```
Comment (strkkk) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/illegalidentifiername/InputIllegalIdentifierNameLambda.java:29]: can we have here case like
```
case var -> .....
case (var) -> .....
```
?

**Comment by nmancus1** [comment]
Github, generate report

**Review by strkkk — APPROVED** [review]

**Comment by strkkk** [comment]
@nmancus1 yes. For some checks it is taking forever to complete and job fails. Probably, it should be removed from default list of projects.

**Comment by nmancus1** [comment]
@strkkk https://github.com/checkstyle/contribution/pull/550

**Review by pbludov — APPROVED** [review]
OK to merge. Note that after https://openjdk.java.net/jeps/302 the underscore will be a valid identifier for a lambda variable name again.

**Comment by rnveach** [comment]
@pbludov Please don't forget to merge PRs or assign the next reviewer.

**Review by rnveach — APPROVED** [review]

**Pull request merged**